manim/utils/ipython_magic.py

```diff
@@ -0,0 +1,115 @@
+"""Utilities for using Manim with IPython (in particular: Jupyter notebooks)"""
+
+import hashlib
+import mimetypes
+import os
+import shutil
+from pathlib import Path
+
+from manim import config, tempconfig
+
+from .._config.main_utils import parse_args
+
+try:
+    from IPython.core.magic import (
+        Magics,
+        magics_class,
+        line_cell_magic,
+        needs_local_scope,
+    )
+    from IPython.display import display, Image, Video
+except ImportError:
+    pass
```

Comment:
should this new dependencies added to `pyproject.toml`?  Maybe add it as a new extras. Users can install it like
```sh
pip install manim[jupyter]
```
or something similar, according to what the extra is called.

Reply:
Good idea! Can you please check whether I did it correctly? db9bfe54